cornell-dti/course-plan
# Mpa Grad Program & Grad Program Front-end: Req Bar

### Summary 

Added the MPA grad program. 
@willespencer integrated the grad program with his front-end changes! 

### Test Plan 
Try adding the MPA grad program. Make sure that at least one undergrad or grad program is selected! 

NEW front-end test plan
1. Try adding just a grad program, and make sure it displays on the requirements bar correctly, and has the color that colleges usually have
2. Try adding a college, 2 minors, and a grad program (but no major) and ensure everything shows correctly on the requirements bar, and that 3 different colors are used
3. Try adding a major back and ensure the grad program has the same color as the college, but other colors are unique.
4. Try any other permutations!

### Notes 

- The concentration requirement for this program is self-check. I was trying to use toggleable requirements but because all of the concentrations are suggestions, I didn't think toggleable reqs really worked so I kept it self-check for now.
- Some of the grad requirements are Yes/No requirements (similar to the Swim Test) so we should add those next sem
- The code to determine whether headers should display is quite complicated, and uses `reqIndex` and `numOfColleges` quite frequently. Left it mostly in-tact but added more to make sure grad programs work, but a refactor would make this code more simple and easy to change in the future.
